lib/wca_i18n/yaml_with_original_hashes.rb

```diff
@@ -0,0 +1,100 @@
+require "psych"
+
+module WcaI18n
+  PLURALIZATION_KEYS = %w(zero one two few many other).freeze
+  TranslatedLeaf = Struct.new(:translated, :original_hash)
+
+  # Re-implement some parts of the ToRuby emitter to inject our TranslatedLeaf where needs be
+  class ToRubyEmitter < Psych::Visitors::ToRuby
+    def self.create(original_hashes_map)
+      class_loader = Psych::ClassLoader.new
+      scanner      = Psych::ScalarScanner.new class_loader
+      new(scanner, class_loader, original_hashes_map)
+    end
+
+    def initialize(ss, cl, original_hashes_map)
+      super(ss, cl)
```

Comment:
I believe we could get rid of the confusing/irrelevant `ss, cl` part this way:
```ruby
def initialize(*args, original_hashes_map)
  super(*args)
  @original_hashes_map = original_hashes_map
end
```
How does it sound?

Reply:
Better :)